core/imds/src/test/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataAsyncClientTest.java

```diff
@@ -0,0 +1,337 @@
+import static com.github.tomakehurst.wiremock.http.Fault.MALFORMED_RESPONSE_CHUNK;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.fail;
+
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.http.Fault;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.retry.backoff.FixedDelayBackoffStrategy;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+import software.amazon.awssdk.imds.Ec2MetadataAsyncClient;
+import software.amazon.awssdk.imds.Ec2MetadataRetryPolicy;
+import software.amazon.awssdk.imds.MetadataResponse;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultEc2MetadataAsyncClientTest {
+
+    private static final String TOKEN_RESOURCE_PATH = "/latest/api/token";
+    private static final String TOKEN_HEADER = "x-aws-ec2-metadata-token";
+    private static final String EC2_METADATA_TOKEN_TTL_HEADER = "x-aws-ec2-metadata-token-ttl-seconds";
+    private static final String EC2_METADATA_ROOT = "/latest/meta-data";
+    private static final String AMI_ID_RESOURCE = EC2_METADATA_ROOT + "/ami-id";
+
+    private static final int DEFAULT_TOTAL_ATTEMPTS = 4;
+
+
+    @Rule
+    public WireMockRule mockMetadataEndpoint = new WireMockRule();
+
+    @Test
+    public void get_successOnFirstTry_shouldNotRetryAndCompleteSuccessfully() {
+        stubFor(put(urlPathEqualTo(TOKEN_RESOURCE_PATH)).willReturn(aResponse().withBody("some-token")));
+        stubFor(get(urlPathEqualTo(AMI_ID_RESOURCE)).willReturn(aResponse().withBody("{}")));
+
+        System.setProperty(SdkSystemSetting.AWS_EC2_METADATA_SERVICE_ENDPOINT.property(),
+                           "http://localhost:" + mockMetadataEndpoint.port());
+        try (Ec2MetadataAsyncClient client =
+                Ec2MetadataAsyncClient.builder()
+                                      .endpoint(URI.create("http://localhost:" + mockMetadataEndpoint.port()))
+                                      .tokenTtl(Duration.ofSeconds(1024))
+                                      .build())
+        {
+            CompletableFuture<MetadataResponse> res = client.get(AMI_ID_RESOURCE);
+            assertThat(res).isNotCompleted();
```

Comment:
`assertThat(res).isNotCompleted();` and `assertThat(res).isCompleted()` seem unnecessary. Same for other test code